src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSubscriber.kt

```diff
@@ -33,7 +35,24 @@ class RabbitMessageGroupBatchSubscriber(
     override fun getDeliveryCounter(): Counter = INCOMING_MSG_GROUP_BATCH_QUANTITY
     override fun getContentCounter(): Counter = INCOMING_MSG_GROUP_QUANTITY
     override fun getProcessingTimer(): Histogram = MSG_GROUP_PROCESSING_TIME
-    override fun extractCountFrom(message: MessageGroupBatch): Int = message.groupsCount
+    override fun extractLabels(batch: MessageGroupBatch): Array<String> {
+        val message = getMessages(batch)[0].messagesList[0]
+
+        return when {
+            message.hasMessage() -> {
+                val messageID = message.message.metadata.id
+                arrayOf(messageID.connectionId.sessionAlias, messageID.direction.name)
```

Comment:
You can extract that into a separate method. And it will look a lot shorter

```kotlin
return when {
            message.hasMessage() -> getLables(message.message.metadata.id)
            message.hasRawMessage() -> getLables(message.rawMessage.metadata.id)
            else -> emptyArray()
        }
```